src/router.ts

```diff
@@ -6,6 +6,8 @@ export interface Route {
 	path: Address[]
 	outputToken: Address
 	outputAmount: BigNumber
+	// the input amounts for each step of the path
+	pathInputAmounts: BigNumber[]
```

Comment:
keep track of the intermediate input token amounts so that the caller can use this to compute the optional minInputAmounts